domain/agentprovisioner/service/service.go

```
@@ -213,11 +214,46 @@ func (s *Service) ContainerConfig(ctx context.Context) (container.Config, error)
 	result.SnapStoreAssertions = modelConfig[config.SnapStoreAssertionsKey]
 	result.SnapStoreProxyID = modelConfig[config.SnapStoreProxyKey]
 	result.SnapStoreProxyURL = modelConfig[config.SnapStoreProxyURLKey]
-	_ = yaml.Unmarshal([]byte(modelConfig[config.CloudInitUserDataKey]), &result.CloudInitUserData)
```

Yikes, why were we ignoring the error. Glad it's fixed.